python/tests/test_ops.py

```diff
@@ -1402,6 +1402,15 @@ def test_stack(self):
         np_c = np.stack([np_a, np_b], axis=1)
         self.assertTrue(np.array_equal(c, np_c))
 
+    def test_linspace(self):
+        a = mx.linspace(0, 10, 5)
+        np_a = np.linspace(0, 10, 5)
+        self.assertTrue(np.array_equal(a, np_a))
+
+        b = mx.linspace(0, 10, 5, dtype=mx.int32)
+        np_b = np.linspace(0, 10, 5, dtype=int)
+        self.assertTrue(np.array_equal(b, np_b))
```

Comment:
Could you add a few more tests here, maybe some where the `start` and `stop` have unusual types, or edge cases like `num=0`?

Reply:
Sorry for the delay getting this merged. I ran into an issue testing irrational number step sizes. For example:

<img width="923" alt="Screenshot 2023-12-18 at 2 37 00 AM" src="https://github.com/ml-explore/mlx/assets/95333017/ccd10319-5da0-4f56-82a6-20e5c884c550">

After some digging, I believe this is an issue with the float precision of `mx.arange` (limitation of numbers being stored as `float32`).

Here's an example to show the floating point issue with `mx.arange`:

<img width="925" alt="Screenshot 2023-12-18 at 2 38 30 AM" src="https://github.com/ml-explore/mlx/assets/95333017/0a37f4d4-9bfe-4e24-af47-5a931014f275">

I also tested against `torch.linspace` and it has similar weird floating point behavior, but the outputs are not consistent.

<img width="922" alt="Screenshot 2023-12-18 at 3 02 22 AM" src="https://github.com/ml-explore/mlx/assets/95333017/17bdf07b-4a2a-4f5d-806d-255a149685ac">

Maybe we need to add `float64` support to `mx.arange`? Let me know your thoughts.

Reply:
@abeleinin that's a good point but at the moment we don't support `mx.float64` and we probably won't have it for a little while. So for testing you can just use `mx.allclose` instead of `array_equal` to make sure values are sufficiently similar.